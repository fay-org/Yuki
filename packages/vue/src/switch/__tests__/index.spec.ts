import { mount } from '@vue/test-utils'
import { Switch, SwitchEvent } from '..'

const Wrapper = {
  components: {
    [Switch.name]: Switch
  },
  data() {
    return {
      val: '',
      checked: 2,
      size: 'medium',
      disabled: false,
      inactive: 3
    }
  },
  template: `<div class="container">
   <fe-switch v-model="val" 
   :checked-value="checked" 
   :size="size" 
   :disabled="disabled" 
   :inactive-value="inactive" />
  </div>`
}

describe('Switch', () => {
  it('should be render as a element', () => {
    const wrapper = mount(Switch)
    expect(() => wrapper.unmount()).not.toThrow()
  })
  it('component props should be work correctly', async () => {
    const wrapper = mount(Wrapper)
    const el = wrapper.find('.fect-switch')
    await el.trigger('click')
    expect(wrapper.vm.val).toBe(2)
    await el.trigger('click')
    expect(wrapper.vm.val).toBe(3)
    await wrapper.setData({ disabled: true, size: 'mini' })
    await el.trigger('click')
    expect(wrapper.emitted('update:modelValue')).toBeFalsy()
    expect(wrapper.html()).toMatchSnapshot()
  })
  it('switch should support control data flow by manual', async () => {
    let state: unknown
    const switchChangeHandler = (e: SwitchEvent) => {
      state = e.target.checked
    }
    const wrapper = mount(Switch, {
      props: {
        value: true,
        onChange: switchChangeHandler
      }
    })
    expect(wrapper.find('.fect-switch--checked').exists()).toBe(true)
    const el = wrapper.find('.fect-switch')
    await el.trigger('click')
    expect(state).toBe(false)
  })
})
