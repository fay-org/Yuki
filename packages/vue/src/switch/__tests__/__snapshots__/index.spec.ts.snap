// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Switch component props should be work correctly 1`] = `
<div class="container"><label class="fect-switch fect-switch--mini fect-switch--disabled"><input class="fect-switch__checkbox" type="checkbox" disabled="">
    <div class="fect-switch__slider"><span class="fect-switch__inner fect-switch__inner--disabled"></span></div>
  </label></div>
`;
